uaclient/config.py

```diff
@@ -665,6 +664,56 @@ def help(self, name):
         response_dict["help"] = help_ent.help_info
         return response_dict
 
+    def process_config(self, assume_yes: bool = False):
+        apt.setup_apt_proxy_with_prompts(
+            self.apt_http_proxy, self.apt_https_proxy, assume_yes=assume_yes
+        )
+
+        suggested_proxy_clearing_cmds = []
+        if snap.is_installed():
```

Comment:
This test is suffecient for presence of snap enabling us to set snap proxy configuration. It is not sufficient for canonical-livepatch though. canonical-livepatch snap can be installed on a non-LTS release (e.g. groovy) and will block any attempt to run any command:
```bash
$ canonical-livepatch status
2021/06/28 12:03:28 error executing status: Livepatchd error: The platform Ubuntu 20.10 is not supported. exiting.
$ echo $?
1
```


For Livepatch, we actually need to check `LivepatchEntitlement.application_status()` to determine enabled or not, or read current service status from the cache via something like:
```python
    livepatch_status = "disabled"
    for service in UAConfig().read_cache("status-cache").get("services", []):
         if service["name"] == "livepatch":
               livepatch_status = service["status"]
               break
```

Only then do we know that the livepatch snap will accept our proxy config attempt.